02-hey-jupyter/README.md

```diff
@@ -0,0 +1,151 @@
+running Jupyter instance on your local system, but if it doesn't, the
+system should print a special URL to copy/paste into your browser:
+   
+```
+$ jupyter notebook
+[I 10:18:00.711 NotebookApp] Serving notebooks from local directory: C:\projects\training\02-hey-jupyter
+[I 10:18:00.711 NotebookApp] 0 active kernels
+[I 10:18:00.711 NotebookApp] The Jupyter Notebook is running at:
+[I 10:18:00.711 NotebookApp] http://localhost:8888/?token=3635b10886fafde18eb8dc781639b068b00299b99ff76aa4
+[I 10:18:00.711 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
+[C 10:18:00.714 NotebookApp]
+
+    Copy/paste this URL into your browser when you connect for the first time,
+    to login with a token:
+        http://localhost:8888/?token=3635b10886fafde18eb8dc781639b068b00299b99ff76aa4
+[I 10:18:00.987 NotebookApp] Accepting one-time-token-authenticated connection from ::1
+```
+
+The interface should show a list of all files in this directory.
+
+### My First Notebook
+
+Open the `scigrid.ipynb` file and execute all the cells. The CSV files
+containing information about terminal nodes (such as substations) and
+transmission lines should appear.
+
+> Aside: The .ipynb extension refers to IPython, or Interactive Python,
+> a predecessor to Jupyter.
+
+### Count Records
+
+Let's determine the number of substations in this dataset. Filter the
+`terminals` data frame for entries with a `typ` of "substation":
+
+```
+substations = terminals[terminals["typ"] == "substation"]
+substations
+```
+
+To determine a count, output `len(substations)` instead.
+
+### Summary Statistics
+
+_pandas_ provides tools for summary statistics. For example, to determine
+the number of branches at each voltage level, we can use:
+
+```
+branches.groupby(by="voltage").size()
+```
+
+Answer the following questions:
+
+1. What is the most common voltage level for substations? For simplicity,
+   ignore stations tha tlist multiple voltage levels.
+
+1. How many unique transmission system operators are captured in this
+   dataset? For simplicity, consider co-owned lines to be a separate
+   operator.
+
+   _Hint_: The results of a grouping for branches and terminals will
+   need to be added together, unless you used a `merge` operation.
+
+1. How many transmission lines are longer than 10km?
+
+1. What are the two most common resistance/km values?
+
+### Derived Values
+
+We can derive new columns based on values of existing columns. For example,
+to determine total resistance of a given line segment, we can use:
+
+```
+branches["r_total"] = branches["r_ohmkm"] * (branches["length_m"] / 1000)
+branches
+```
+
+1. Which line has the largest total resistance?
+
+1. What are the lines with the 10 largest X/R ratios? 10 smallest?
+
+1. Which line has the highest total impedance? (magnitude only)
+   
+   _Hint_: recall that `Z = R + jX` and `|Z| = sqrt(R^2 + X^2)`
```

Comment:
This hint probably isn't needed and might seem pedantic to a seasoned TPE

Reply:
Good point. Removed.